samples/matter/common/src/diagnostic/diagnostic_logs_provider.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+
+#include "persistent_storage/persistent_storage_util.h"
+#include "util/finite_map.h"
+
+#include <app/clusters/diagnostic-logs-server/DiagnosticLogsProviderDelegate.h>
+#include <lib/core/CHIPError.h>
+
+namespace Nrf::Matter
+{
+
+class DiagnosticLogProvider : public chip::app::Clusters::DiagnosticLogs::DiagnosticLogsProviderDelegate {
+public:
+	static inline DiagnosticLogProvider &GetInstance() { return sInstance; }
+
+	CHIP_ERROR GetCrashLogs(chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog);
+	CHIP_ERROR ClearCrashLogs();
+
+	/* DiagnosticsLogsProviderDelegate Interface */
+	CHIP_ERROR StartLogCollection(chip::app::Clusters::DiagnosticLogs::IntentEnum intent,
+				      chip::app::Clusters::DiagnosticLogs::LogSessionHandle &outHandle,
+				      chip::Optional<uint64_t> &outTimeStamp,
+				      chip::Optional<uint64_t> &outTimeSinceBoot) override;
+	CHIP_ERROR EndLogCollection(chip::app::Clusters::DiagnosticLogs::LogSessionHandle sessionHandle) override;
+	CHIP_ERROR CollectLog(chip::app::Clusters::DiagnosticLogs::LogSessionHandle sessionHandle,
+			      chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog) override;
+	size_t GetSizeForIntent(chip::app::Clusters::DiagnosticLogs::IntentEnum intent) override;
+	CHIP_ERROR GetLogForIntent(chip::app::Clusters::DiagnosticLogs::IntentEnum intent,
+				   chip::MutableByteSpan &outBuffer, chip::Optional<uint64_t> &outTimeStamp,
+				   chip::Optional<uint64_t> &outTimeSinceBoot) override;
+
+private:
+	/* The maximum number of the simultaneous sessions */
+	constexpr static uint16_t kMaxLogSessionHandle =
+		CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_MAX_SIMULTANEOUS_SESSIONS;
+	constexpr static uint16_t kInvalidLogSessionHandle = -1;
+
+	static_assert(kMaxLogSessionHandle < chip::app::Clusters::DiagnosticLogs::kInvalidLogSessionHandle);
```

Comment:
You could also add `range` to the Kconfig, so it would not be possible to set it in a wrong way.

Reply:
But the value from `chip::app::Clusters::DiagnosticLogs::kInvalidLogSessionHandle`may change in the future. I guess we cannot set `range` to the Kconfig.

Reply:
Ok, now I'm confused :) I didn't realize the `kInvalidLogSessionHandle` is something from the Matter core. Then why do we need `kInvalidLogSessionHandle = -1` above?

Reply:
Because I forgot to remove it :)

Reply:
Ok, then it's clear for me now.